src/Adapter/PlatformServices/Services/TestSourceHost.cs

```diff
@@ -252,16 +252,19 @@ private void SetContext(string source)
         _currentDirectory = Environment.CurrentDirectory;
         try
         {
+            // If the source is in the format of an assembly qualified name, then calls to
+            // Path.GetDirectoryName will return empty string. But if we use Path.GetFullPath first
+            // then directory resolution works properly.
+            var dirName = Path.GetDirectoryName(Path.GetFullPath(source));
```

Comment:
This behavior fixes some of the tests. I don't think it has any bad consequences compared to existing code.